apps/examples/testcase/le_tc/network/tc_net_tcp.c

```diff
@@ -0,0 +1,318 @@
+	result = tcp_process_refused_data(pcb);
+	TC_ASSERT_EQ("tc_net_tcp_process_refused_data_p", result, ERR_OK);
+	TC_SUCCESS_RESULT();
+}
+
+/**
+* @statitcase			: tc_net_tcp_setprio_n
+* @brief				:
+* @scenario				:
+* @apicovered			: tcp_setprio
+* @precondition			:
+* @postcondition		:
+*/
+
+static void tc_net_tcp_setprio_n(void)
+{
+	struct tcp_pcb *pcb;
+	u8_t prio = 129;
+
+	if (!(pcb = (struct tcp_pcb *)malloc(sizeof(struct tcp_pcb))))
+		return;
+	tcp_setprio(pcb, prio);
+	TC_SUCCESS_RESULT();
+}
+
+/**
+* @statitcase			: tc_net_tcp_setprio_p
+* @brief				:
+* @scenario				:
+* @apicovered			: tcp_setprio
+* @precondition			:
+* @postcondition		:
+*/
+static void tc_net_tcp_setprio_p(void)
+{
+	struct tcp_pcb *pcb;
+	u8_t prio = 5;
+	if (!(pcb = (struct tcp_pcb *)malloc(sizeof(struct tcp_pcb))))
+		return ;
+	tcp_setprio(pcb, prio);
+	TC_SUCCESS_RESULT();
+}
+
+/**
+* @statitcase			: tc_net_tcp_kill_prio_p
+* @brief				:
+* @scenario				:
+* @apicovered			: tcp_kill_prio
+* @precondition			:
+* @postcondition		:
+*/
+static void tc_net_tcp_kill_prio_p(void)
+{
+	u8_t prio;
+	prio = 5;
+	tcp_kill_prio(prio);
+	TC_SUCCESS_RESULT();
+}
+
+/**
+* @statitcase			: tc_net_tcp_kill_prio_n
+* @brief				:
+* @scenario				:
+* @apicovered			: tcp_kill_prio
+* @precondition			:
+* @postcondition		:
+*/
+static void tc_net_tcp_kill_prio_n(void)
+{
+	u8_t prio;
+	prio = 0;
+	tcp_kill_prio(prio);
+	TC_SUCCESS_RESULT();
+}
+
+/**
+* @statitcase			: tc_net_tcp_recv_null_n
+* @brief				:
+* @scenario				:
+* @apicovered			: tcp_recv_null
+* @precondition			:
+* @postcondition		:
+*/
+static void tc_net_tcp_recv_null_n(void)
+{
+	struct tcp_pcb *pcb = NULL;
+	struct pbuf *p = NULL;
+	err_t result;
+	result = tcp_recv_null(NULL, pcb, p, 0);
+	TC_ASSERT_EQ("tc_net_tcp_recv_null_n", result, ERR_VAL);
+	TC_SUCCESS_RESULT();
+}
+
+/**
+* @statitcase			: tc_net_tcp_recv_null_p
+* @brief				:
+* @scenario				:
+* @apicovered			: tcp_recv_null
+* @precondition			:
+* @postcondition		:
+*/
+static void tc_net_tcp_recv_null_p(void)
+{
+	struct tcp_pcb *pcb = NULL;
+	struct pbuf *p = NULL;
+	err_t result;
+	if (!(pcb = (struct tcp_pcb *)malloc(sizeof(struct tcp_pcb))))
+		return;
+	if (!(p = (struct pbuf *)malloc(sizeof(struct pbuf))))
+		return;
+	result = tcp_recv_null(NULL, pcb, p, 0);
+	TC_ASSERT_EQ("tc_net_tcp_recv_null_p", result, ERR_OK);
+	TC_SUCCESS_RESULT();
+}
+
+/**
+* @statitcase			: tcp_kill_timewait_p
+* @brief				:
+* @scenario				:
+* @apicovered			: tcp_kill_timewait
+* @precondition			:
+* @postcondition		:
+*/
+static void tc_net_tcp_kill_timewait_p(void)
+{
+	tcp_kill_timewait();
+	TC_SUCCESS_RESULT();
+}
+
+/**
+* @statitcase			: tcp_debug_state_str_p
+* @brief				:
+* @scenario				:
+* @apicovered			: tcp_debug_state_str
+* @precondition			:
+* @postcondition		:
+*/
+static void tc_net_tcp_debug_state_str_p(void)
+{
+	const char *result;
+	result = tcp_debug_state_str(0);
+	TC_ASSERT_EQ("tc_net_tcp_debug_state_str_p", result, "CLOSED");
```

Comment:
Again string is compared with result var !! 

#define TC_ASSERT_EQ_RETURN(api_name, var, ref, ret) \
{ \
        if ((var) != (ref)) { \
                printf("\n[%s][Line : %d] FAIL, %s : Values (%s == 0x%x) and (%s == 0x%x) are not equal\n", __func__, __LINE__, api_name, #var, (int)(var), #ref, (int)(ref)); \
                total_fail++; \
                return ret; \
        } \
}

#define TC_ASSERT_EQ(api_name, var, ref) TC_ASSERT_EQ_RETURN(api_name, var, ref, )
 So is it correct ?